### PR TITLE
fix(generate): stop treating ai@6 raw-text output echo as parsed schema output

### PR DESCRIPTION
**Pull request by murdore**
## Problem

Since ai@6 resolves `output ?? text()` **inside** `generateText`, a result produced *without* an output spec no longer throws from the `experimental_output` getter — it **echoes the raw model text as a string**. Two NeuroLink paths produce such results while a schema is active:

1. the structured-output **fallback retry** (after `NoObjectGeneratedError` / tools↔schema conflict / schema-complexity errors), and
2. the **tools↔schema exclusion** path (Gemini, native Anthropic surface).

`formatEnhancedResult` treats any defined `experimental_output` as the AI-SDK's parsed+validated object, so on those paths it:

- sets `structuredData` to the raw **text string**, and
- `JSON.stringify`s it into `content` — double-encoding every response.

For a **non-empty** response the facade's JSON-string-literal unwrap accidentally repairs the damage downstream. For an **EMPTY completion** it survives to the caller as content `'""'` (the literal two characters) with `structuredData ''` — so consumers' empty-response handling never fires and users see a literal `""`. This was found in production-style E2E testing of curator's empty-response retry (its retry predicate never fired on the schema path).

## Fix (3 changes)

| Site | Change |
|---|---|
| `GenerationHandler.formatEnhancedResult` | Detect the raw-text **echo** (`typeof experimental_output === 'string' && === generateResult.text`) and route it through text-mode coercion instead of serialising it as schema output |
| `GenerationHandler` `coerceTextMode` scalar path | A JSON-encoded empty string (`'""'`) is an **empty completion**, not a recovered scalar — normalize to `''` content, no `structuredData`, + WARN |
| `neurolink.ts` facade scalar path | Same guard for provider-native `generate()` overrides (Vertex / Google AI / Anthropic native) |

Genuine `Output.object` results are unaffected: their `experimental_output` is a parsed object (or a parsed string ≠ raw text), never the raw-text echo.

## Verification (stubbed Anthropic API, wire-level)

| Case | Before | After |
|---|---|---|
| Empty completion + schema (fallback path) | content `'""'`, structuredData `''` | content `''`, structuredData unset ✅ |
| Non-JSON first attempt → fallback returns valid JSON | repaired only by accidental facade unwrap | parsed object + single-encoded content at the formatter ✅ |
| Model literally emits `""` as its completion | content `'""'`, structuredData `''` | content `''`, structuredData unset ✅ |
| Empty completion, no schema | `''` | `''` (no regression) ✅ |

- `tsc --noEmit` clean
- `continuous-test-suite-structured-coerce` / `coerce-nested-unwrap` / `json` — all PASS
- prettier clean; eslint: 0 errors (7 pre-existing max-lines warnings on untouched functions)

🤖 Investigated & authored with Claude Code

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed structured-output handling for empty responses.
  * Prevented echoed model text from being incorrectly treated as parsed structured data.
  * Avoided unwanted JSON-stringified content in generated results.
  * Added warnings when empty or invalid structured-output recovery occurs.

### CODE REVIEW

**Review by Tara-ag — APPROVED** [review]
## Review Summary

**Files reviewed:** 2 (`src/lib/core/modules/GenerationHandler.ts`, `src/lib/neurolink.ts`)

**New issues raised this run:** 0

**Decision:** Approve

The fix correctly identifies and routes the ai@6 raw-text echo away from the structured-output path, and the empty-JSON-string normalization addresses the production empty-response bug described in the PR. The changes are surgical, well-commented, and consistent with the existing coercion patterns in `coerceJsonToSchema`.

No blocking criteria are met: no secrets, no architectural rule violations, no breaking public SDK API changes, and no new MAJOR correctness issues. The existing unresolved review comment from the Copilot reviewer requesting a regression test for the literal `""` path is noted; it is not duplicated here per review protocol, but should be resolved before merge.

File: src/lib/core/modules/GenerationHandler.ts [hunk 1]
```diff
@@ -859,6 +859,17 @@ export class GenerationHandler {
...
+          // A JSON-encoded empty string is an EMPTY completion, not a
+          // recovered scalar — normalize to a true empty ('' content, no
+          // structuredData) so callers' empty-response handling fires
+          // instead of a literal '""' reaching the user.
+          logger.warn(
+            "[GenerationHandler] schema requested but the model returned an empty JSON string; normalizing to empty content",
+            { provider: this.providerName, model: this.modelName },
+          );
+          return "";
+        }
```
[the remark this answers is not in the document]
Comment (murdore) [reply src/lib/core/modules/GenerationHandler.ts:872]: Added in 56c1d017 — `test/continuous-test-suite-schema-empty-normalization.ts` (pure suite, no API) drives `formatEnhancedResult` directly through all five shapes: empty raw-text echo (fallback result), literal `""` completion, non-empty echo (must coerce, not double-encode), genuine `Output.object` object (preserved), and a parsed string ≠ raw text (`z.string()` schema, stays structured). Verified it fails 3/5 against the pre-fix code and passes 5/5 with the fix.

**Review by Tara-ag — APPROVED** [review]
## Review Summary

Reviewed 3 changed files. No new blocking issues found.

### Files reviewed
- `src/lib/core/modules/GenerationHandler.ts`
- `src/lib/neurolink.ts`
- `test/continuous-test-suite-schema-empty-normalization.ts`

### Assessment
- The raw-text echo detection in `formatEnhancedResult` correctly identifies the ai@6 `output ?? text()` fallback case and routes it through text coercion instead of double-encoding.
- The empty-JSON-string (`""`) normalization is applied consistently in both the `GenerationHandler` path and the facade scalar path in `neurolink.ts`, preventing the literal `'""'` content bug.
- The regression test suite covers the key shapes: empty raw-text echo, literal `""`, non-empty echo, genuine `Output.object`, and parsed string ≠ raw text.
- No CLAUDE.md critical-rule violations, security issues, or public SDK API breakage detected.

### Existing comment
The existing Copilot review thread requesting a regression test for the literal `""` path appears addressed by the added `test/continuous-test-suite-schema-empty-normalization.ts`.

**Decision: Approve.**

File: test/continuous-test-suite-schema-empty-normalization.ts
Comment (Tara-ag) [inline test/continuous-test-suite-schema-empty-normalization.ts:1]: 💡 The new regression suite is well-scoped and covers the critical edge cases. Per `CONTRIBUTING.md`, new `continuous-test-suite-*.ts` files should also get a matching `test:<name>` script in `package.json` (or be imported by the main `test/continuous-test-suite.ts` orchestrator) so this runs in CI and doesn’t silently regress. Consider adding e.g.:

```json
"test:schema-empty-normalization": "npx tsx test/continuous-test-suite-schema-empty-normalization.ts"
```

and wiring it into the relevant aggregate script.

**Review by Tara-ag — APPROVED** [review]
## Review Summary

**Files reviewed:** 3
**New issues raised:** 1 (MINOR)
**Blocking issues:** 0

### Assessment

The fix correctly addresses the ai@6 raw-text echo regression:

- `GenerationHandler.formatEnhancedResult` now detects when `experimental_output` is the raw model text echo and routes it through text-mode coercion instead of treating it as parsed schema output.
- The JSON-encoded empty string (`""`) is normalized to a true empty completion with `structuredData` unset, which fixes the literal `'""'` content bug.
- The same guard is applied in `neurolink.ts` for provider-native `generate()` overrides.
- A pure, no-API regression suite covers the five critical shapes: empty raw-text echo, literal `""`, non-empty echo coercion, genuine `Output.object`, and parsed string ≠ raw text.

No hardcoded secrets, security vulnerabilities, architectural rule violations, or public SDK API breaks were found.

### Minor note

The new regression suite is not yet wired into `package.json` or the main `test/continuous-test-suite.ts` orchestrator, so it will not run in CI. I left an inline suggestion on the test file to add a matching script. This is non-blocking.

Approving — the core fix is sound and the regression coverage is good.

**Review by Tara-ag — APPROVED** [review]
## Review Summary

**Files reviewed:** 3
**New issues raised:** 0
**Existing unresolved comments:** 1 minor suggestion (package.json wiring for the new test suite — already noted by Tara-ag)

### Assessment

The fix correctly addresses the ai@6 raw-text echo behavior in structured-output fallback paths:

- `GenerationHandler.formatEnhancedResult` now detects the raw-text echo (`experimental_output` string identical to `generateResult.text`) and routes it through text-mode coercion instead of serializing it as parsed schema output.
- The scalar coercion paths in both `GenerationHandler` and `neurolink.ts` normalize a JSON-encoded empty string (`'""'`) to a true empty completion, preventing the literal two-character string from reaching callers.
- The regression suite covers the critical shapes: empty raw-text echo, literal `'""'`, non-empty echo coercion, genuine `Output.object` objects, and parsed strings that differ from raw text.

### Checked against project rules

- **CLAUDE.md Critical Rule 5 (backward compatibility):** No breaking changes to the public SDK API; the change only corrects edge-case output for schema fallback paths.
- **Security:** No secrets, credentials, or PII exposure in the new logs; `logger.warn` only includes provider/model identifiers.
- **Type safety / correctness:** No `any`/non-null assertions introduced in production code; the logic correctly distinguishes raw-text echoes from genuine parsed scalar strings.
- **Testing:** New pure suite added in the `continuous-test-suite-*.ts` pattern.

### Note

The existing unresolved comment from Tara-ag about adding a matching `test:schema-empty-normalization` script in `package.json` remains unaddressed. That is a valid CI/wiring suggestion but is non-blocking; it can be followed up in this PR or a fast-follow.

Approving — the blocking bug is fixed and the change is low-risk.
